src/daft-core/src/array/ops/utf8.rs

```diff
@@ -772,6 +773,239 @@ impl Utf8Array {
+                return Ok(val.chars().take(length).collect());
+            }
+            let fillchar = if fillchar.is_empty() {
+                return Err(DaftError::ComputeError(
+                    "Error in rpad: empty pad character".to_string(),
+                ));
+            } else if fillchar.chars().count() > 1 {
+                return Err(DaftError::ComputeError(format!(
+                    "Error in rpad: {} is not a valid pad character",
+                    fillchar.len()
+                )));
+            } else {
+                fillchar.chars().next().unwrap()
+            };
+            let fillchar =
+                std::iter::repeat(fillchar).take(length.saturating_sub(val.chars().count()));
+            Ok(val.chars().chain(fillchar).collect())
+        }
+
+        let self_iter = create_broadcasted_str_iter(self, expected_size);
+        let padchar_iter = create_broadcasted_str_iter(padchar, expected_size);
+        let result = match length.len() {
+            1 => {
+                let len = length.get(0).unwrap();
+                let len: usize = NumCast::from(len).ok_or_else(|| {
+                    DaftError::ComputeError(format!(
+                        "Error in rpad: failed to cast length as usize {len}"
+                    ))
+                })?;
+                let arrow_result = self_iter
+                    .zip(padchar_iter)
+                    .map(|(val, padchar)| match (val, padchar) {
+                        (Some(val), Some(padchar)) => Ok(Some(rpad_str(val, len, padchar)?)),
+                        _ => Ok(None),
+                    })
+                    .collect::<DaftResult<arrow2::array::Utf8Array<i64>>>()?;
+
+                Utf8Array::from((self.name(), Box::new(arrow_result)))
+            }
+            _ => {
+                let length_iter = length.as_arrow().iter();
+                let arrow_result = self_iter
+                    .zip(length_iter)
+                    .zip(padchar_iter)
+                    .map(|((val, len), padchar)| match (val, len, padchar) {
+                        (Some(val), Some(len), Some(padchar)) => {
+                            let len: usize = NumCast::from(*len).ok_or_else(|| {
+                                DaftError::ComputeError(format!(
+                                    "Error in rpad: failed to cast length as usize {len}"
+                                ))
+                            })?;
+                            Ok(Some(rpad_str(val, len, padchar)?))
+                        }
+                        _ => Ok(None),
+                    })
+                    .collect::<DaftResult<arrow2::array::Utf8Array<i64>>>()?;
+
+                Utf8Array::from((self.name(), Box::new(arrow_result)))
+            }
+        };
+
+        assert_eq!(result.len(), expected_size);
+        Ok(result)
+    }
+
+    pub fn lpad<I>(&self, length: &DataArray<I>, padchar: &Utf8Array) -> DaftResult<Utf8Array>
```

Comment:
Aside from `rpad_str` and `lpad_str`, to my understanding the logic is pretty much the same between the lpad and rpad. I'm wondering if it's possible to consolidate it somehow? Perhaps doing some sort of refactor of `parse_inputs` as you mentioned before would be good, or moving things into another function that takes in either `rpad_str` or `lpad_str` as arguments.